packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -166,66 +161,41 @@ def __new__(mcs, name: str, bases: Tuple, namespace: Dict, **kwargs: Any) -> Typ
                 f"class {name} must inherit from {BaseTxPayload.__name__}"
             )
         new_cls = cast(Type[BaseTxPayload], new_cls)
-
-        transaction_type = str(mcs._get_field(new_cls, "transaction_type"))
-        mcs._validate_transaction_type(transaction_type, new_cls)
         # remember association from transaction type to payload class
-        mcs.transaction_type_to_payload_cls[transaction_type] = new_cls
+        _metaclass_registry_key = f"{new_cls.__module__}.{new_cls.__name__}"  # type: ignore
+        mcs.registry[_metaclass_registry_key] = new_cls
```

Comment:
```suggestion
        if not _metaclass_registry_key in mcs.registry:
            mcs.registry[_metaclass_registry_key] = new_cls
        else:
            enforce(mcs.registry[_metaclass_registry_key] == new_cls, "Duplicate key {_metaclass_registry_key} detected! Ensure you only define one Payload class per skill!")
```

Reply:
How would that even be possible? If you redefine a class with the same name in the same module, it will overwrite and only the last one can be imported from the module where ever you try to use it.
```
_MetaPayload.registry.items()
('packages.valory.skills.hello_world_abci.payloads.RegistrationPayload', <class 'packages.valory.skills.hello_world_abci.payloads.RegistrationPayload'>)
('packages.valory.skills.hello_world_abci.payloads.CollectRandomnessPayload', <class 'packages.valory.skills.hello_world_abci.payloads.CollectRandomnessPayload'>)
('packages.valory.skills.hello_world_abci.payloads.PrintMessagePayload', <class 'packages.valory.skills.hello_world_abci.payloads.PrintMessagePayload'>)
('packages.valory.skills.hello_world_abci.payloads.SelectKeeperPayload', <class 'packages.valory.skills.hello_world_abci.payloads.SelectKeeperPayload'>)
('packages.valory.skills.hello_world_abci.payloads.ResetPayload', <class 'packages.valory.skills.hello_world_abci.payloads.ResetPayload'>)
```

Reply:
Yes correct - not needed. I thought we had scoped to skill, not module